crates/rattler_repodata_gateway/src/sparse/mod.rs

```diff
@@ -136,7 +137,17 @@ impl SparseRepoData {
 
         // Iterate over the list of packages that still need to be processed.
         while let Some(next_package) = pending.pop_front() {
+            let mut found_in_channel = None;
             for (i, repo_data) in repo_data.iter().enumerate() {
+                // If package was found in other channel, skip this repodata
+                if strict_channel_priority {
+                    if let Some(url) = found_in_channel.clone() {
+                        if repo_data.channel.base_url != url {
+                            continue;
+                        }
+                    }
+                }
```

Comment:
This can be done much shorter:

```suggestion
                if found_in_channel.map_or(false, |c| c != &repo_data.channel.base_url) {
                    continue;
                }
```

You dont need to check the `strict_channel_priority` because the `found_in_channel` is only set if that is true.

Reply:
That is much better!